nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/commands/DeleteJvmOptions.java

```diff
@@ -158,12 +159,22 @@ public Object run(JavaConfig jc) throws PropertyVetoException, TransactionFailur
     }
     */
     //@ForTimeBeing :)
-    private void deleteX(final JvmOptionBag bag, final List<String> toRemove, final ActionReport.MessagePart part) throws Exception {
+    private void deleteX(final JvmOptionBag bag, final List<String> toRemoveOptions, final ActionReport.MessagePart part) throws Exception {
         SingleConfigCode<JvmOptionBag> scc = (JvmOptionBag bag1) -> {
             List<String> jvmopts = new ArrayList<>(bag1.getJvmRawOptions());
             int orig = jvmopts.size();
-            // using new JvmOption(option).toString() (instead op option directly) to make sure the correct formatting is applied.
-            boolean removed = jvmopts.removeIf(option -> toRemove.contains(new JvmOption(option).toString()));
+            boolean removed = false;
+            Iterator<String> iter = jvmopts.iterator();
+            for (String toRemoveOption : toRemoveOptions) {
+                String option = new JvmOption(toRemoveOption).option;
+                while (iter.hasNext()) {
```

Comment:
This construct fails in some cases when multiple options are removed.  Iterator is only initialised once, so you can have the situation that the second `toRemoveOption` value is already in the part of the iterator which has been traversed already.

Reply:
@rdebusscher, you are right. It was a dumb mistake on my part